source/Octopus.Tentacle.Tests.Integration/Support/OctopusPackageDownloader.cs

```diff
@@ -17,56 +18,99 @@ public class OctopusPackageDownloader
+                        logger.Information($"Downloading {downloadUrl} ({totalLength} bytes)");
 
-                        var read = await contentStream.ReadAsync(buffer, 0, buffer.Length);
-                        while (read != 0)
+                        var sw = new Stopwatch();
+                        sw.Start();
+                        using (Stream contentStream = await response.Content.ReadAsStreamAsync(),
+                               fileStream = new FileStream(
+                                   filePath,
+                                   FileMode.Create,
+                                   FileAccess.Write,
+                                   FileShare.None,
+                                   8192,
+                                   true))
                         {
-                            await fileStream.WriteAsync(buffer, 0, read);
 
-                            if (totalLength.HasValue && sw.ElapsedMilliseconds >= TimeSpan.FromSeconds(7).TotalMilliseconds)
+                            var buffer = new byte[8192];
+
+                            var read = await contentStream.ReadAsync(buffer, 0, buffer.Length);
+                            while (read != 0)
                             {
-                                var percentRead = totalRead * 1.0 / totalLength.Value * 100;
-                                logger.Information($"Downloading Completed {percentRead}%");
-                                sw.Reset();
-                                sw.Start();
+                                await fileStream.WriteAsync(buffer, 0, read);
+
+                                if (totalLength.HasValue && sw.ElapsedMilliseconds >= TimeSpan.FromSeconds(7).TotalMilliseconds)
+                                {
+                                    var percentRead = totalRead * 1.0 / totalLength.Value * 100;
+                                    logger.Information($"Downloading Completed {percentRead}%");
+                                    sw.Reset();
+                                    sw.Start();
+                                }
+
+                                read = await contentStream.ReadAsync(buffer, 0, buffer.Length);
+                                totalRead += read;
                             }
+                            
+                            totalTime.Stop();
 
-                            read = await contentStream.ReadAsync(buffer, 0, buffer.Length);
-                            totalRead += read;
+                            logger.Information("Download Finished in {totalTime}", totalTime.ElapsedMilliseconds);
                         }
-
-                        logger.Information("Download Finished");
                     }
                 }
             }
+            catch (Exception e)
+            {
+                throw new Exception($"Failure to download: {downloadUrl}. After {totalTime.Elapsed.TotalSeconds} seconds we only downloaded, {totalRead}", e);
```

Comment:
These logs don't make it to the build server :(, so instead make the exception contain information already in the logs.